### PR TITLE
Add zxing code39reader patch

### PR DESCRIPTION
**Pull request by steerlink**
@davegreco @ShaneSmit 

I apologize for all the whitespace changes.

### CODE REVIEW

File: contrib/src/zxing/code39reader.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/core/src/zxing/oned/Code39Reader.cpp b/core/src/zxing/oned/Code39Reader.cpp
+index 9719781..3944c7f 100644
+--- a/core/src/zxing/oned/Code39Reader.cpp
++++ b/core/src/zxing/oned/Code39Reader.cpp
+@@ -133,7 +133,7 @@ Ref<Result> Code39Reader::decodeRow(int rowNumber, Ref<BitArray> row) {
+   int whiteSpaceAfterEnd = nextStart - lastStart - lastPatternSize;
```
Comment (ShaneSmit) [inline contrib/src/zxing/code39reader.patch]: This line causes me precedence stress.  Can we get some parenthesis here?

Comment (steerlink) [inline contrib/src/zxing/code39reader.patch]: I think it is pretty clear. Left to right associativity. Plus this line isn't getting modified. It is original source. Lines 9 and 10 have the change which is simply a bit shift.

Comment (ShaneSmit) [inline contrib/src/zxing/code39reader.patch]: Right, if it's coming from an "original" source, then leave it.  But if we wrote it, then I would push for parenthesis.

**Comment by ShaneSmit** [comment]
:shipit:

File: contrib/src/zxing/code39reader.patch (hunk 1 above)
Comment (davegreco) [inline contrib/src/zxing/code39reader.patch]: Yeah, this was just what git spit out as a patch. I also brought up that it had a little bit of noise. However @steerlink pointed out that the exact coordinates are mapped with the change the the rest is just for easy human readable context/consumption.

**Pull request merged**